data_safe_haven/external/api/azure_cli.py

```diff
@@ -81,3 +81,25 @@ def confirm(self) -> None:
             raise typer.Exit(1)
 
         self._confirmed = True
+
+    def group_id_from_name(self, group_name: str) -> str:
+        """Get ID for an Entra ID group that this user is permitted to view."""
+        try:
+            result = subprocess.check_output(
+                [self.path, "ad", "group", "list", "--display-name", group_name],
+                stderr=subprocess.PIPE,
+                encoding="utf8",
+            )
+        except subprocess.CalledProcessError as exc:
+            msg = f"Error reading groups from Azure CLI.\n{exc.stderr}"
+            raise DataSafeHavenAzureError(msg) from exc
+
+        try:
+            result_dict = json.loads(result)
+            return str(result_dict[0]["id"])
+        except json.JSONDecodeError as exc:
+            msg = f"Unable to parse Azure CLI output as JSON.\n{result}"
+            raise DataSafeHavenAzureError(msg) from exc
+        except (IndexError, KeyError) as exc:
+            msg = f"Group '{group_name}' was not found in Azure CLI."
+            raise DataSafeHavenAzureError(msg) from exc
```

Comment:
This feels fairly complex/fragile.

I'm not sure the gain of giving a name rather than id is worth it.

Reply:
I'm not sure it's as fragile as it looks. I don't think the JSONDecodeError is very likely, for example, so could be merged with the IndexError and KeyError catch.

IMO, seeing "DSH Test Admins" in the Context file is more readable/understandable than seeing the GUID and having to lookup whether it's the correct one.

Reply:
```suggestion
    def group_id_from_name(self, group_name: str) -> str:
        """Get ID for an Entra ID group that this user is permitted to view."""
        try:
            result = subprocess.check_output(
                [self.path, "ad", "group", "list", "--display-name", group_name],
                stderr=subprocess.PIPE,
                encoding="utf8",
            )
            result_dict = json.loads(result)
            return str(result_dict[0]["id"])
         except (subprocess.CalledProcessError, json.JSONDecodeError, IndexError, KeyError) as exc:
            msg = f"Group '{group_name}' was not found in Azure CLI."
            raise DataSafeHavenAzureError(msg) from exc
```

Reply:
I think I prefer the original, distinguishes between "Azure CLI didn't give me JSON" and "The thing you are looking for doesn't exist".

We can leave it if we think the benefit is there.

In general, I don't like the subprocess calls as we have to support ABI compatibility with Azure CLI. This feels like a place where we don't need one because the group name and id have the same meaning.

Reply:
I personally prefer the name rather than the GUID, as it's much more obvious to the user if it's incorrect

Reply:
OK, let's leave this as it is then